src/Lykke.blue.Service.ReferralLinks/Controllers/ReferralLinksController.cs

```diff
@@ -0,0 +1,480 @@
+            }
+
+            var balance = clientBalances.FirstOrDefault(x => x.AssetId == asset.Id)?.Balance;
+
+            if (!balance.HasValue || balance.Value < request.Amount)
+            {
+                return await LogAndReturnBadRequest(request, ControllerContext, Phrases.InvalidTreesAmount);
+            }
+
+            var referralLink = await _referralLinksService.CreateGiftCoinsLink(request);
+
+            await LogInfo(request, ControllerContext, referralLink.ToJson());
+
+            return Created(uri: $"api/referralLinks/{referralLink.Id}", value: new RequestRefLinkResponse { RefLinkId = referralLink.Id, RefLinkUrl = referralLink.Url });
+        }
+
+
+        /// <summary>
+        /// Request invitation referral link.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        [HttpPost("invitationLinks")]
+        [SwaggerOperation("RequestInvitationReferralLink")]
+        [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType(typeof(RequestRefLinkResponse), (int)HttpStatusCode.Created)]
+        [ProducesResponseType(typeof(RequestRefLinkResponse), (int)HttpStatusCode.OK)]
+        public async Task<IActionResult> RequestInvitationReferralLink([FromBody] InvitationReferralLinkRequest request)
+        {
+            if (request == null)
+            {
+                return await LogAndReturnBadRequest("", ControllerContext, Phrases.InvalidRequest);
+            }
+
+            if (String.IsNullOrEmpty(request.SenderClientId))
+            {
+                return await LogAndReturnBadRequest(request, ControllerContext, Phrases.InvalidSenderClientId);
+            }
+
+            var invitationLinkAlreadyCreated = _referralLinksService.GetInvitationLinksForSenderId(request.SenderClientId).ToList();
+            if (invitationLinkAlreadyCreated.Count == 1)
+            {
+
+                await LogInfo(request, ControllerContext, $"Invitation link already exists for SenderClientId {request.SenderClientId}. RefLinkId {invitationLinkAlreadyCreated.First().Id}");
+                return Ok(new RequestRefLinkResponse { RefLinkUrl = invitationLinkAlreadyCreated.First().Url, RefLinkId = invitationLinkAlreadyCreated.First().Id });
+            }
+            if (invitationLinkAlreadyCreated.Count > 1)
+            {
+                return await LogAndReturnInternalServerError(request, ControllerContext, $"There are multiple invitation links created for client {request.SenderClientId}! ");
+            }
+
+            if (!_referralLinksService.InvitationLinkForSenderIdExists(request.SenderClientId))
```

Comment:
This check is redundant since we have the similar check at line 233.

Reply:
Check was put as an extra measure for ensuring there is not link created for client.

Reply:
Fixed